radis/spectrum/compare.py

```diff
@@ -261,174 +261,129 @@ def get_residual(
-    resampled through get_diff on 's1' range 
-
-    Implementation of ``L2`` norm::
-
-        np.sqrt((dI**2).sum())/len(dI)
-    
-    Implementation of ``L1`` norm::
+        ``'mean'`` will normalize by the mean amplitude value. Ex::
+        
+            get_residual(s_exp, s_calc, normalize=True, normalize_how == "area") 
         
-        np.abs(dI).sum()/len(dI)
-
     See Also
     --------
-
     :func:`~radis.spectrum.compare.get_diff`, 
     :func:`~radis.spectrum.compare.get_ratio`, 
     :func:`~radis.spectrum.compare.get_distance`, 
     :func:`~radis.spectrum.compare.plot_diff`, 
     :func:`~radis.spectrum.compare.get_residual_integral`, 
     :meth:`~radis.spectrum.spectrum.compare_with` 
     """
+    from numpy.linalg import norm
+
+    if Iunit == "default":
+        Iunit = s1.units[var]
+    if wunit == "default":
+        wunit = s1.get_waveunit()
 
+    # Get data
+    # ----
     if normalize:
-        from radis.spectrum.operations import multiply
-
-        if isinstance(normalize, tuple):
-            wmin, wmax = normalize
-            w1, I1 = s1.get(
-                var, copy=False, wunit=s1.get_waveunit()
-            )  # (faster not to copy)
-            b = (w1 > wmin) & (w1 < wmax)
-            if normalize_how == "max":
-                norm1 = np.nanmax(I1[b])
-            elif normalize_how == "mean":
-                norm1 = np.nanmean(I2[b])
-            elif normalize_how == "area":
-                norm1 = np.abs(nantrapz(I1[b], w1[b]))
-            else:
-                raise ValueError(
-                    "Unexpected `normalize_how`: {0}".format(normalize_how)
-                )
-            # now normalize s2. Ensure we use the same unit system!
-            w2, I2 = s2.get(var, Iunit=s1.units[var], wunit=s1.get_waveunit())
-            b = (w2 > wmin) & (w2 < wmax)
-            if normalize_how == "max":
-                norm2 = np.nanmax(I2[b])
-            elif normalize_how == "mean":
-                norm2 = np.nanmean(I2[b])
-            elif normalize_how == "area":
-                norm2 = np.abs(nantrapz(I2[b], w2[b]))
-            else:
-                raise ValueError(
-                    "Unexpected `normalize_how`: {0}".format(normalize_how)
-                )
-            s1 = multiply(s1, 1 / norm1, var=var)
-            s2 = multiply(s2, 1 / norm2, var=var)
+        # copy before modifying directly in spectrum
+        s1 = s1.copy()
+        s2 = s2.copy()
+        w1, I1 = s1.get(var, wunit=wunit, copy=False)
+        w2, I2 = s2.get(var, wunit=wunit, copy=False)
+        if normalize_how == "max":
+            norm1 = np.nanmax(I1) 
+            norm2 = np.nanmax(I2)
+        elif normalize_how == "mean":
+            norm1 = np.nanmean(I1) 
+            norm2 = np.nanmean(I2)
+        elif normalize_how == "area":
+            norm1 = np.abs(nantrapz(I1, w1))
+            norm2 = np.abs(nantrapz(I2, w2))

```

Comment:
Since 0.9.26 I think we can now use the following, quite simple syntax :

```suggestion
        s1 = s1.normalize(inplace=False, normalize_how=normalize_how, wrange=normalize)
        s2 = s2.normalize(inplace=False, normalize_how=normalize_how, wrange=normalize)
```

What do you think ?